### PR TITLE
 test machine with spec securityProfile encryption at rest as "true"

### PR DESCRIPTION
**Pull request by miyadav**
@jhou1 @sunzhaohua2 PTAL , whenever you get time
Auto test to validate changes by - https://bugzilla.redhat.com/show_bug.cgi?id=1900454

```
.
.
.
    
    [11:23:46] INFO> Exit Status: 0
    [11:23:46] INFO> Shell Commands: oc delete machines encrypt-at-rest-39639-6bq5t --wait=false --config=/root/workdir/d97593744bfd-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
    machine.machine.openshift.io "encrypt-at-rest-39639-6bq5t" deleted
    
    [11:23:48] INFO> Exit Status: 0
    [11:23:50] INFO> oc get machines encrypt-at-rest-39639-6bq5t --output=yaml --config=/root/workdir/d97593744bfd-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
    [11:25:49] INFO> After 45 iterations and 121 seconds:
    
    STDERR:
    Error from server (NotFound): machines.machine.openshift.io "encrypt-at-rest-39639-6bq5t" not found
    
    [11:25:49] INFO> Shell Commands: oc get machines encrypt-at-rest-39639-6bq5t --output=yaml --config=/root/workdir/d97593744bfd-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
    
    STDERR:
    Error from server (NotFound): machines.machine.openshift.io "encrypt-at-rest-39639-6bq5t" not found
    
    [11:25:51] INFO> Exit Status: 1
    [11:25:51] INFO> Shell Commands: rm -r -f -- /root/workdir/d97593744bfd-
    
    [11:25:52] INFO> Exit Status: 0
    [11:25:53] INFO> === End After Scenario: Implement defaulting machineset values for azure, Examples (#2) ===

2 scenarios (2 passed)
30 steps (30 passed)
5m27.997s
[11:25:53] INFO> === At Exit ===
```

### CODE REVIEW

File: testdata/cloud/ms-azure/ms_encrypt_at_rest.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - effect: "NoSchedule"
+        key: "mapi"
+        value: "mapi_test"
+      providerSpec:
+        value:
+          location:
+          osDisk:
+            diskSizeGB: 128
+          securityProfile:
+                  encryptionAtHost: true
```
Comment (sunzhaohua2) [inline testdata/cloud/ms-azure/ms_encrypt_at_rest.yaml:34]: seems there are extra spaces in this line, can you confirm? @miyadav

Comment (miyadav) [reply testdata/cloud/ms-azure/ms_encrypt_at_rest.yaml]: done

File: features/machine/machine.feature [hunk 2]
```diff
@@ -370,28 +370,50 @@ Feature: Machine features testing
...
     Then the step should succeed
+  
+    And I wait up to 120 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource | machine |
+    Then the step should succeed
+    And the output should contain:
+      | Running |
+    """
```
Comment (sunzhaohua2) [inline features/machine/machine.feature:403]: These steps always work, as no machine name is specified, we can use below steps to get the same result after  line 405.
```
    And I wait up to 120 seconds for the steps to pass:
    """
    Then the expression should be true> machine(cb.machine_latest).phase(cached: false) == "Running"
    """
```

Comment (miyadav) [reply features/machine/machine.feature]: Agreed , will need this fix for one more test  , will do it with a seperate branch , this one is corrected

**Pull request merged**